### PR TITLE
dev-lua/lua-openssl: new package

### PR DESCRIPTION
**Pull request by jirutka**

### CODE REVIEW

File: dev-lua/lua-openssl/lua-openssl-0.4.1.ebuild [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+		-i Makefile || die "failed to filter Makefile"
+}
+
+src_compile() {
+	local pkgconfig=$(tc-getPKG_CONFIG)
+	emake \
+		LUA_CFLAGS="$($pkgconfig --cflags $LUAV)" \
+		LUA_LIBS="$($pkgconfig --libs $LUAV)" \
+		LUA_LIBDIR="$($pkgconfig --variable INSTALL_CMOD $LUAV)" \
+		|| die "emake failed"
```
Comment (hasufell) [inline dev-lua/lua-openssl/lua-openssl-0.4.1.ebuild]: `|| die` not needed in EAPPI=5 for `emake`

File: dev-lua/lua-openssl/lua-openssl-0.4.1.ebuild [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+}
+
+src_install() {
+	emake \
+		LUA_LIBDIR="${D}$($(tc-getPKG_CONFIG) --variable INSTALL_CMOD $LUAV)" \
+		install || die "emake install failed"
```
Comment (hasufell) [inline dev-lua/lua-openssl/lua-openssl-0.4.1.ebuild]: `|| die` not needed in EAPPI=5 for `emake`

File: dev-lua/lua-openssl/lua-openssl-0.4.1.ebuild [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+	virtual/pkgconfig"
+
+use luajit && LUAV=luajit || LUAV=lua
+
+src_prepare() {
+	sed -r \
+		-e 's|^CFLAGS(\s*)=|CFLAGS\1+=|' \
+		-e 's|^LDFLAGS(\s*)=|LDFLAGS\1+=|' \
+		-e "s|^CC=.*|CC=$(tc-getCC) \$(CFLAGS) -Ideps|" \
+		-i Makefile || die "failed to filter Makefile"
```
Comment (hasufell) [inline dev-lua/lua-openssl/lua-openssl-0.4.1.ebuild]: looks like we want a patch here. why so much sedding?

File: dev-lua/lua-openssl/lua-openssl-0.4.1.ebuild [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+inherit toolchain-funcs
+
+DESCRIPTION="OpenSSL binding for Lua"
+HOMEPAGE="https://github.com/zhaozg/lua-openssl"
+SRC_URI="https://github.com/zhaozg/lua-openssl/archive/${PV}.tar.gz -> lua-openssl-${PV}.tar.gz"
+
+LICENSE="MIT"
+SLOT="0"
+KEYWORDS="~amd64 ~x86"
+IUSE="+luajit"
```
Comment (hasufell) [inline dev-lua/lua-openssl/lua-openssl-0.4.1.ebuild]: not sure if `+luajit` is a good idea. Is the luajit version preferable for any reason? If not, just do `luajit`

File: dev-lua/lua-openssl/lua-openssl-0.4.1.ebuild [hunk 5]
```diff
@@ -0,0 +1,48 @@
...
+
+RDEPEND="
```
Comment (hasufell) [inline dev-lua/lua-openssl/lua-openssl-0.4.1.ebuild]: dependency on `dev-libs/openssl:0` is completely missing... while at it, also [add libressl support](https://github.com/gentoo/libressl/wiki/Transition-plan)

File: dev-lua/lua-openssl/lua-openssl-0.4.1.ebuild (hunk 4 above)
Comment (hasufell) [inline dev-lua/lua-openssl/lua-openssl-0.4.1.ebuild]: afais, upstream has default `luajit`, so I guess it makes sense

File: dev-lua/lua-openssl/lua-openssl-0.4.1.ebuild (hunk 1 above)
Comment (jirutka) [inline dev-lua/lua-openssl/lua-openssl-0.4.1.ebuild]: Didn’t know that, thanks.

File: dev-lua/lua-openssl/lua-openssl-0.4.1.ebuild (hunk 4 above)
Comment (jirutka) [inline dev-lua/lua-openssl/lua-openssl-0.4.1.ebuild]: Well, luajit is much faster, but there’s no other special reason for luajit in this package, so okay.

File: dev-lua/lua-openssl/lua-openssl-0.4.1.ebuild (hunk 5 above)
Comment (jirutka) [inline dev-lua/lua-openssl/lua-openssl-0.4.1.ebuild]: Oh, you’re right, I totally forgot to add it.

Comment (jirutka) [inline dev-lua/lua-openssl/lua-openssl-0.4.1.ebuild]: I haven’t tested it with libressl and there’s no note about libressl compatibility in lua-openssl. Are you sure that we can add it?

File: dev-lua/lua-openssl/lua-openssl-0.4.1.ebuild (hunk 3 above)
Comment (jirutka) [inline dev-lua/lua-openssl/lua-openssl-0.4.1.ebuild]: I prefer sed for smaller changes because it’s much simpler to maintain and more tolerant to changes in the source file.

File: dev-lua/lua-openssl/lua-openssl-0.4.1.ebuild (hunk 5 above)
Comment (hasufell) [inline dev-lua/lua-openssl/lua-openssl-0.4.1.ebuild]: seems libressl support will require a patch: https://github.com/zhaozg/lua-openssl/pull/81

File: dev-lua/lua-openssl/lua-openssl-0.4.1.ebuild (hunk 3 above)
Comment (hasufell) [inline dev-lua/lua-openssl/lua-openssl-0.4.1.ebuild]: Gentoo generally prefers clean patches, because they don't silently fail.

File: dev-lua/lua-openssl/lua-openssl-0.4.1.ebuild [hunk 6]
```diff
@@ -0,0 +1,49 @@
...
+		LUA_CFLAGS="$($pkgconfig --cflags $LUAV)" \
+		LUA_LIBS="$($pkgconfig --libs $LUAV)" \
+		LUA_LIBDIR="$($pkgconfig --variable INSTALL_CMOD $LUAV)"
+}
+
+src_install() {
+	emake \
+		LUA_LIBDIR="${D}$($(tc-getPKG_CONFIG) --variable INSTALL_CMOD $LUAV)" \
+		install
+	dodoc README.md
```
Comment (hasufell) [inline dev-lua/lua-openssl/lua-openssl-0.4.1.ebuild]: looks like you want "einstalldocs"

File: dev-lua/lua-openssl/metadata.xml [hunk 7]
```diff
@@ -0,0 +1,11 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<!DOCTYPE pkgmetadata SYSTEM "http://www.gentoo.org/dtd/metadata.dtd">
+<pkgmetadata>
+  <maintainer>
```
Comment (hasufell) [inline dev-lua/lua-openssl/metadata.xml]: unless you have a specific gentoo developer proxying for you, you need to add
```xml
<herd>proxy-maintainers</herd>
```

File: dev-lua/lua-openssl/lua-openssl-0.4.1.ebuild [hunk 8]
```diff
@@ -0,0 +1,49 @@
...
+
+RDEPEND="
+	luajit? ( dev-lang/luajit:2 )
+	!luajit? ( >=dev-lang/lua-5.1 )"
+DEPEND="${RDEPEND}
+	libressl? ( dev-libs/libressl )
+	!libressl? ( dev-libs/openssl:0 )
+	virtual/pkgconfig"
+
+use luajit && LUAV=luajit || LUAV=lua
```
Comment (hasufell) [inline dev-lua/lua-openssl/lua-openssl-0.4.1.ebuild]: That is absolutely not allowed. No use-conditionals in global scope. You can do that in pkg_setup or src_prepare

File: dev-lua/lua-openssl/lua-openssl-0.4.1.ebuild [hunk 9]
```diff
@@ -0,0 +1,49 @@
...
+inherit toolchain-funcs
+
+DESCRIPTION="OpenSSL binding for Lua"
+HOMEPAGE="https://github.com/zhaozg/lua-openssl"
+SRC_URI="https://github.com/zhaozg/lua-openssl/archive/${PV}.tar.gz -> lua-openssl-${PV}.tar.gz"
+
+LICENSE="MIT"
+SLOT="0"
+KEYWORDS="~amd64 ~x86"
+IUSE="luajit libressl"
```
Comment (hasufell) [inline dev-lua/lua-openssl/lua-openssl-0.4.1.ebuild]: for libressl compatibility we will need a patch like https://patch-diff.githubusercontent.com/raw/zhaozg/lua-openssl/pull/81.patch but we should have upstream take a look over it, so remove libressl support for now

File: dev-lua/lua-openssl/lua-openssl-0.4.1.ebuild [hunk 10]
```diff
@@ -0,0 +1,49 @@
...
+
+RDEPEND="
+	luajit? ( dev-lang/luajit:2 )
+	!luajit? ( >=dev-lang/lua-5.1 )"
+DEPEND="${RDEPEND}
+	libressl? ( dev-libs/libressl )
+	!libressl? ( dev-libs/openssl:0 )
```
Comment (hasufell) [inline dev-lua/lua-openssl/lua-openssl-0.4.1.ebuild]: openssl is not just a build-time dependency, lua-openssl _links_ to it

File: dev-lua/lua-openssl/metadata.xml (hunk 7 above)
Comment (jirutka) [inline dev-lua/lua-openssl/metadata.xml]: Like this…?

```xml
<pkgmetadata>
  <maintainer>
    <herd>proxy-maintainers</herd>
  </maintainer>
  <upstream>
    <remote-id type="github">zhaozg/lua-openssl</remote-id>
  </upstream>
</pkgmetadata>
```

File: dev-lua/lua-openssl/lua-openssl-0.4.1.ebuild [hunk 11]
```diff
@@ -0,0 +1,49 @@
...
+
+EAPI=5
+
+inherit toolchain-funcs
+
+DESCRIPTION="OpenSSL binding for Lua"
+HOMEPAGE="https://github.com/zhaozg/lua-openssl"
+SRC_URI="https://github.com/zhaozg/lua-openssl/archive/${PV}.tar.gz -> lua-openssl-${PV}.tar.gz"
+
+LICENSE="MIT"
```
Comment (hasufell) [inline dev-lua/lua-openssl/lua-openssl-0.4.1.ebuild]: according to https://github.com/zhaozg/lua-openssl/blob/master/LICENSE it also needs `openssl` in the license field

File: dev-lua/lua-openssl/lua-openssl-0.4.1.ebuild (hunk 10 above)
Comment (jirutka) [inline dev-lua/lua-openssl/lua-openssl-0.4.1.ebuild]: Eh, of course, I often confuse RDEPEND and DEPEND.

File: dev-lua/lua-openssl/metadata.xml (hunk 7 above)
Comment (hasufell) [inline dev-lua/lua-openssl/metadata.xml]: no, see https://gitweb.gentoo.org/data/dtd.git/tree/metadata.dtd
```xml
<?xml version="1.0" encoding="UTF-8"?>
<!DOCTYPE pkgmetadata SYSTEM "http://www.gentoo.org/dtd/metadata.dtd">
<pkgmetadata>
  <herd>proxy-maintainers</herd>
  <maintainer>
    <email>jakub@jirutka.cz</email>
    <name>Jakub Jirutka</name>
  </maintainer>
  <upstream>
    <remote-id type="github">zhaozg/lua-openssl</remote-id>
  </upstream>
</pkgmetadata>
```

File: dev-lua/lua-openssl/lua-openssl-0.4.1.ebuild (hunk 6 above)
Comment (jirutka) [inline dev-lua/lua-openssl/lua-openssl-0.4.1.ebuild]: I’ve never seen this command and according to information I’ve found, this will be added in EAPI 6…?

Comment (hasufell) [inline dev-lua/lua-openssl/lua-openssl-0.4.1.ebuild]: until it's part of PMS it is available via `eutils.eclass`

File: dev-lua/lua-openssl/lua-openssl-0.4.1.ebuild (hunk 3 above)
Comment (jirutka) [inline dev-lua/lua-openssl/lua-openssl-0.4.1.ebuild]: Okay then, I’ve replaced it with patch.

File: dev-lua/lua-openssl/lua-openssl-0.4.1.ebuild (hunk 11 above)
Comment (jirutka) [inline dev-lua/lua-openssl/lua-openssl-0.4.1.ebuild]: Their license is very confused. There’s some lua-openssl License (what’s that?), Lua License, OpenSSL License and PHP License (WTF?) in the license file. However, the first line of the readme says that it’s MIT-licensed. :confused:

Comment (hasufell) [inline dev-lua/lua-openssl/lua-openssl-0.4.1.ebuild]: well, then we have to add php too

Comment (jirutka) [inline dev-lua/lua-openssl/lua-openssl-0.4.1.ebuild]: And what about Lua license and that strange lua-openssl license?

Comment (hasufell) [inline dev-lua/lua-openssl/lua-openssl-0.4.1.ebuild]: those are MIT

Comment (jirutka) [inline dev-lua/lua-openssl/lua-openssl-0.4.1.ebuild]: Okay, updated.

File: dev-lua/lua-openssl/lua-openssl-0.4.1.ebuild [hunk 12]
```diff
@@ -0,0 +1,46 @@
+# Copyright 1999-2015 Gentoo Foundation
+# Distributed under the terms of the GNU General Public License v2
+# $Id$
+
+EAPI=5
+
+inherit base eutils toolchain-funcs
```
Comment (hasufell) [inline dev-lua/lua-openssl/lua-openssl-0.4.1.ebuild]: base.eclass is deprecated, it should not be used any longer

File: dev-lua/lua-openssl/lua-openssl-0.4.1.ebuild [hunk 13]
```diff
@@ -0,0 +1,46 @@
...
+
+DESCRIPTION="OpenSSL binding for Lua"
+HOMEPAGE="https://github.com/zhaozg/lua-openssl"
+SRC_URI="https://github.com/zhaozg/lua-openssl/archive/${PV}.tar.gz -> lua-openssl-${PV}.tar.gz"
+
+LICENSE="MIT openssl PHP-3"
+SLOT="0"
+KEYWORDS="~amd64 ~x86"
+IUSE="luajit"
```
Comment (hasufell) [inline dev-lua/lua-openssl/lua-openssl-0.4.1.ebuild]: use `repoman full` to check your ebuild:
```
  IUSE.invalid [fatal]          1
   dev-lua/lua-openssl/lua-openssl-0.4.1.ebuild: luajit
```

File: dev-lua/lua-openssl/lua-openssl-0.4.1.ebuild [hunk 14]
```diff
@@ -0,0 +1,46 @@
...
+
+RDEPEND="
+	luajit? ( dev-lang/luajit:2 )
+	!luajit? ( >=dev-lang/lua-5.1 )
```
Comment (hasufell) [inline dev-lua/lua-openssl/lua-openssl-0.4.1.ebuild]:
```
dependency.missingslot        1
   dev-lua/lua-openssl/lua-openssl-0.4.1.ebuild: RDEPEND: '>=dev-lang/lua-5.1' matches more than one slot, please specify an explicit slot and/or use the := or :* slot operator
```

I guess we want SLOT `:0`

File: dev-lua/lua-openssl/lua-openssl-0.4.1.ebuild (hunk 12 above)
Comment (jirutka) [inline dev-lua/lua-openssl/lua-openssl-0.4.1.ebuild]: Well, but what other eclass handles `$PATCHES` variable? Or should I use `epatch` inside `src_prepare` instead?

Comment (hasufell) [inline dev-lua/lua-openssl/lua-openssl-0.4.1.ebuild]: yes

File: dev-lua/lua-openssl/lua-openssl-0.4.1.ebuild [hunk 15]
```diff
@@ -0,0 +1,46 @@
...
+DOCS=( "README.md" )
+
+pkg_setup() {
+	use luajit && LUAV=luajit || LUAV=lua
+}
+
+src_compile() {
+	local pkgconfig=$(tc-getPKG_CONFIG)
+	emake \
+		CC="$(tc-getCC) -g \$(CFLAGS) -Ideps" \
```
Comment (hasufell) [inline dev-lua/lua-openssl/lua-openssl-0.4.1.ebuild]: `-g` is a debug flag and should not be set by ebuilds

File: dev-lua/lua-openssl/lua-openssl-0.4.1.ebuild [hunk 16]
```diff
@@ -0,0 +1,46 @@
...
+
+RDEPEND="
+	luajit? ( dev-lang/luajit:2 )
+	!luajit? ( >=dev-lang/lua-5.1 )
+	dev-libs/openssl:0"
+DEPEND="${RDEPEND}
+	virtual/pkgconfig"
+
+PATCHES=( "${FILESDIR}/${P}-Makefile.patch" )
+DOCS=( "README.md" )
```
Comment (hasufell) [inline dev-lua/lua-openssl/lua-openssl-0.4.1.ebuild]: the DOCS variable is obsolete

File: dev-lua/lua-openssl/files/lua-openssl-0.4.1-Makefile.patch [hunk 17]
```diff
@@ -0,0 +1,51 @@
+diff --git a/Makefile b/Makefile
```
Comment (hasufell) [inline dev-lua/lua-openssl/files/lua-openssl-0.4.1-Makefile.patch]: add patch description, also see https://dev.gentoo.org/~vapier/clean-patches

File: dev-lua/lua-openssl/lua-openssl-0.4.1.ebuild (hunk 13 above)
Comment (jirutka) [inline dev-lua/lua-openssl/lua-openssl-0.4.1.ebuild]: Fixed.

File: dev-lua/lua-openssl/files/lua-openssl-0.4.1-Makefile.patch (hunk 17 above)
Comment (jirutka) [inline dev-lua/lua-openssl/files/lua-openssl-0.4.1-Makefile.patch]: Okay, I’ve updated it. However, to be honest, I’m not sure if this patch is really needed/useful. I just saw this in other Lua ebuilds. I _think_ that the rationale behind it is to respect flags defined in the system environment. (I have little experience with C stuff)

Comment (hasufell) [inline dev-lua/lua-openssl/files/lua-openssl-0.4.1-Makefile.patch]: This patch should even go upstream.

**Comment by hasufell** [comment]
@rafaelmartins maybe have a look too?

File: dev-lua/lua-openssl/files/lua-openssl-0.4.1-Makefile.patch [hunk 18]
```diff
@@ -0,0 +1,53 @@
...
+--- Makefile
++++ Makefile
+@@ -22,30 +22,30 @@ SYS := $(shell gcc -dumpmachine)
+ 
+ ifneq (, $(findstring linux, $(SYS)))
+ # Do linux things
+-LDFLAGS		    = -fPIC -lrt -ldl
++LDFLAGS		    += -fPIC -lrt -ldl
+ OPENSSL_LIBS	?= $(shell pkg-config openssl --libs) 
+ OPENSSL_CFLAGS	?= $(shell pkg-config openssl --cflags)
```
Comment (hasufell) [inline dev-lua/lua-openssl/files/lua-openssl-0.4.1-Makefile.patch]: I just noticed this... this will break cross compiling and maybe other things, because `pkg-config` is hardcoded.

The Makefile should do:
```
PKG_CONFIG ?= pkg-config
[...]
OPENSSL_CFLAGS	?= $(shell $(PKG_CONFIG) openssl --cflags)
[...]
```

File: dev-lua/lua-openssl/lua-openssl-0.4.1.ebuild [hunk 19]
```diff
@@ -0,0 +1,44 @@
...
+		LUA_CFLAGS="$($pkgconfig --cflags $LUAV)" \
+		LUA_LIBS="$($pkgconfig --libs $LUAV)" \
+		LUA_LIBDIR="$($pkgconfig --variable INSTALL_CMOD $LUAV)"
+}
+
+src_install() {
+	emake \
+		LUA_LIBDIR="${D}$($(tc-getPKG_CONFIG) --variable INSTALL_CMOD $LUAV)" \
+		install
+	einstalldocs README.md
```
Comment (hasufell) [inline dev-lua/lua-openssl/lua-openssl-0.4.1.ebuild]: `einstalldocs` alone is fine

Comment (mgorny) [inline dev-lua/lua-openssl/lua-openssl-0.4.1.ebuild]: Let me rephrase that for you: `einstalldocs` doesn't take any arguments

File: dev-lua/lua-openssl/files/lua-openssl-0.4.1-Makefile.patch (hunk 18 above)
Comment (mgorny) [inline dev-lua/lua-openssl/files/lua-openssl-0.4.1-Makefile.patch]: And then you need to pass `PKG_CONFIG=...` to make too.

Comment (hasufell) [inline dev-lua/lua-openssl/files/lua-openssl-0.4.1-Makefile.patch]: exactly

File: dev-lua/lua-openssl/lua-openssl-0.4.1.ebuild [hunk 20]
```diff
@@ -0,0 +1,44 @@
...
+
+LICENSE="MIT openssl PHP-3"
+SLOT="0"
+KEYWORDS="~amd64 ~x86"
+IUSE="luajit"
+
+RDEPEND="
+	luajit? ( dev-lang/luajit:2 )
+	!luajit? ( >=dev-lang/lua-5.1:0 )
+	dev-libs/openssl:0"
```
Comment (hasufell) [inline dev-lua/lua-openssl/lua-openssl-0.4.1.ebuild]: good news: https://github.com/zeroc-ice/ice/commit/5974de7290be5097960a117321fc87fa8fb607d4

Comment (hasufell) [inline dev-lua/lua-openssl/lua-openssl-0.4.1.ebuild]: oops, I mixed something up here :dancer:

**Comment by rafaelmartins** [comment]
other than what already commented by @hasufell and @mgorny it looks good

**Comment by mgorny** [comment]
@jirutka, please address the mentioned issues, and if possible get yourself a nice PGP key if you don't have one already (https://wiki.gentoo.org/wiki/Project:Gentoo-keys/Generating_GLEP_63_based_OpenPGP_keys) and sign the updated commit ;-).

**Comment by jirutka** [comment]
@mgorny I’ll fix it soon (tonight or tomorrow). This commit is already PGP signed:

```
commit ba7aeceba586349a1749f4a5e901c70b208dfee2
gpg: Signature made Sat Oct  3 20:03:30 2015 CEST using RSA key ID 35C69BBA
gpg: Good signature from "Jakub Jirutka <jakub@jirutka.cz>" [ultimate]
Author: Jakub Jirutka <jakub@jirutka.cz>
Date:   Sat Oct 3 16:36:35 2015 +0200

    dev-lua/lua-openssl: new package
```

**Comment by mgorny** [comment]
Good to hear! It's sad GitHub doesn't indicate that directly.

File: dev-lua/lua-openssl/lua-openssl-0.4.1.ebuild (hunk 19 above)
Comment (jirutka) [inline dev-lua/lua-openssl/lua-openssl-0.4.1.ebuild]: Well, then I need to use `DOCS` variable, but @hasufell [told me](https://github.com/gentoo/gentoo/pull/149#discussion_r41088887) that it’s obsolete. How else would `einstalldocs` know what files are docs?

Comment (hasufell) [inline dev-lua/lua-openssl/lua-openssl-0.4.1.ebuild]: > How else would einstalldocs know what files are docs?

it just knows ;) just see the eclass, it's simple bash

Comment (jirutka) [inline dev-lua/lua-openssl/lua-openssl-0.4.1.ebuild]: Aha, mea culpa, I haven’t noticed that it has some defaults!

btw I know eclasses, I even [wrote one](https://github.com/cvut/gentoo-overlay/blob/master/eclass/tomcat.eclass). :)

**Comment by jirutka** [comment]
I’ve fixed all the mentioned issues, commit is PGP signed, so I guess we’re ready for merge! :smile_cat:

**Comment by mgorny** [comment]
Oh no, and I was going to go to sleep!

File: dev-lua/lua-openssl/files/lua-openssl-0.4.1-Makefile.patch (hunk 17 above)
Comment (mgorny) [inline dev-lua/lua-openssl/files/lua-openssl-0.4.1-Makefile.patch]: Well, for our needs the patch will work. However, for upstream I'd improve it a bit. In particular:

1. Don't put CFLAGS in CC, fix the whole thing to respect all variables properly ;-).
2. Don't append to variables like CFLAGS, it won't work if you do like `make CFLAGS='-foo'`, I think. Instead, use extra variables.

But that's future suggestions in case you decide to do sth about it.

File: dev-lua/lua-openssl/lua-openssl-0.4.1.ebuild [hunk 21]
```diff
@@ -0,0 +1,45 @@
...
+# Distributed under the terms of the GNU General Public License v2
+# $Id$
+
+EAPI=5
+
+inherit eutils toolchain-funcs
+
+DESCRIPTION="OpenSSL binding for Lua"
+HOMEPAGE="https://github.com/zhaozg/lua-openssl"
+SRC_URI="https://github.com/zhaozg/lua-openssl/archive/${PV}.tar.gz -> lua-openssl-${PV}.tar.gz"
```
Comment (mgorny) [inline dev-lua/lua-openssl/lua-openssl-0.4.1.ebuild]: Gotcha! `lua-openssl-${PV}` -> `${P}`?

File: dev-lua/lua-openssl/lua-openssl-0.4.1.ebuild [hunk 22]
```diff
@@ -0,0 +1,45 @@
...
+
+LICENSE="MIT openssl PHP-3"
+SLOT="0"
+KEYWORDS="~amd64 ~x86"
+IUSE="luajit"
+
+RDEPEND="
+	luajit? ( dev-lang/luajit:2 )
+	!luajit? ( >=dev-lang/lua-5.1:0 )
```
Comment (mgorny) [inline dev-lua/lua-openssl/lua-openssl-0.4.1.ebuild]: This is not ready for slotted Lua. This is not strictly necessary, I guess, and I doubt I can help you with that :).

Comment (mgorny) [inline dev-lua/lua-openssl/lua-openssl-0.4.1.ebuild]: OTOH slotted Lua is masked like forever and nobody's working on it.

File: dev-lua/lua-openssl/lua-openssl-0.4.1.ebuild (hunk 21 above)
Comment (jirutka) [inline dev-lua/lua-openssl/lua-openssl-0.4.1.ebuild]: Ah, of course. Fixed now!

File: dev-lua/lua-openssl/metadata.xml [hunk 23]
```diff
@@ -0,0 +1,16 @@
...
+		<email>jakub@jirutka.cz</email>
+		<name>Jakub Jirutka</name>
+	</maintainer>
+	<upstream>
+		<remote-id type="github">zhaozg/lua-openssl</remote-id>
+	</upstream>
+	<use>
+		<flag name="luajit">Use <pkg>dev-lang/luajit</pkg> instead of
+			<pkg>dev-lang/lua</pkg></flag>
+	</use>
```
Comment (mgorny) [inline dev-lua/lua-openssl/metadata.xml]: I'm pretty sure the tag order is incorrect. Please verify against the DTD.

File: dev-lua/lua-openssl/lua-openssl-0.4.1.ebuild (hunk 22 above)
Comment (hasufell) [inline dev-lua/lua-openssl/lua-openssl-0.4.1.ebuild]: slot `:0` is not slotted, what's the problem?

Comment (mgorny) [inline dev-lua/lua-openssl/lua-openssl-0.4.1.ebuild]: Future versions.

File: dev-lua/lua-openssl/metadata.xml (hunk 23 above)
Comment (jirutka) [inline dev-lua/lua-openssl/metadata.xml]: Hm, you’re right, `upstream` should be after `use`. Fixed.

Comment (mgorny) [inline dev-lua/lua-openssl/metadata.xml]: Oh, nevermind. It seems that DTD was changed to allow any order.

Comment (mgorny) [inline dev-lua/lua-openssl/metadata.xml]: Though the usual order is herd/maintainer/use/upstream.

File: dev-lua/lua-openssl/lua-openssl-0.4.1.ebuild (hunk 22 above)
Comment (hasufell) [inline dev-lua/lua-openssl/lua-openssl-0.4.1.ebuild]: Especially since nobody knows what they are doing there we should specify the slot we have tested against. If they break a slot, it's their responsibility to fix reverse deps.

**Comment by mgorny** [comment]
All from me. Please let me know when you're done (again) and I'll merge it.

**Comment by jirutka** [comment]
I’m done already. ;) Just don’t know what to do with that slotted Lua dependency…

File: dev-lua/lua-openssl/lua-openssl-0.4.1.ebuild (hunk 22 above)
Comment (mgorny) [inline dev-lua/lua-openssl/lua-openssl-0.4.1.ebuild]: Yes, I know. Just saying that slot :0 is basically deprecated but they never finished the newer slots… which is also why we're still at lua-5.1.

**Comment by mgorny** [comment]
Don't worry about it. Lua's so broken in Gentoo ti doesn't really matter ;-P.

**Comment by jirutka** [comment]
Uh, why it’s broken?

**Comment by mgorny** [comment]
Long story short, people started working on slotting Lua, and never finished.

**Comment by mgorny** [comment]
And if I recall correctly, Lua upstream expects people to bundle or statically link Lua into packages and we diverge by providing a system-wide copy. But this also means we'll eventually need to provide multiple versions simultaneously, that's why it needed to be slotted. But if we do that, we suddenly need to hack everything to select Lua versions correctly.

**Comment by hasufell** [comment]
> Uh, why it’s broken?

Because it is hacked downstream (like the pkg-config file and soname). But that's partly because upstream is uncollaborative.

I started using bundled lua when available because of all the mess.

**Comment by hasufell** [comment]
@jirutka libressl patch was merged upstream! https://github.com/zhaozg/lua-openssl/commit/edef7a3c47981e95ae40ccf42bd553317e6998a1
